.github/workflows/windows.yml

```diff
@@ -10,28 +10,35 @@ jobs:
     runs-on: ${{matrix.os}}
     strategy:
       matrix:
-        # https://github.com/actions/virtual-environments.
         os: [windows-2022]
-        platform: [x64]
-        toolset: [v142]
+        generator: ["Ninja", "Visual Studio 17 2022"]
         standard: [17, 20, 23]
-        build_type: [Debug]
+        # BUG: Ninja + MSVC + Modules fails to build
+        exclude:
+          - {generator: "Ninja", standard: 20 }
+          - {generator: "Ninja", standard: 23 }
 
     steps:
     - uses: actions/checkout@v3
+    - uses: ilammy/msvc-dev-cmd@v1
 
     - name: Create Build Environment
       run: cmake -E make_directory ${{runner.workspace}}/build
 
     - name: Configure
       working-directory: ${{runner.workspace}}/build
-      run: cmake -A ${{matrix.platform}} -T ${{matrix.toolset}} -DCMAKE_CXX_STANDARD=${{matrix.standard}} -DCMAKE_BUILD_TYPE=${{matrix.build_type}} ${env:GITHUB_WORKSPACE}/example
+      env:
+        CC: cl
+      run: cmake -G "${{matrix.generator}}" -DCMAKE_CXX_STANDARD=${{matrix.standard}} -DCMAKE_BUILD_TYPE=Debug ${env:GITHUB_WORKSPACE}/example
 
     - name: Build
       working-directory: ${{runner.workspace}}/build
-      run: cmake --build . --config ${{matrix.build_type}} --parallel
+      run: cmake --build . --config Debug --parallel
 
     - name: Test
       env:
+        FILE_PATH: ${{fromJSON('["build/Debug/main.exe", "build/main.exe"]')[matrix.generator == 'Ninja']}}
         CHECK_STRING: ${{fromJSON('["Modules are late to the party", "Hello, modules"]')[matrix.standard > 17]}}
-      run: $(${{runner.workspace}}/build/${{matrix.build_type}}/main.exe | Out-String) -match "${env:CHECK_STRING}"
+      run: |
+        $result = (& ${{runner.workspace}}\${env:FILE_PATH} | Out-String) -match "${env:CHECK_STRING}"
+        if(!$result) { Write-Error "Bad output" }
```

Comment:
I think we should avoid scripting in yaml. I suggest moving this logic elsewhere, if possible to the binary itself or some python or even cmake script.

Reply:
>or even cmake script.

![image](https://user-images.githubusercontent.com/704382/236560423-930ccd3f-afff-4684-8c1e-2fcebb1ce509.png)

Reply:
Rewritten, now both Linux and Windows use the same CMake script. I put it in `.github/workflows` to avoid confusing people browsing the `example` folder.